<launch>

    <!-- ========================= -->
    <!-- ARGS (opcionales)         -->
    <!-- ========================= -->
    <arg name="serial_port" default="/dev/ttyACM1"/>
    <arg name="baudrate" default="115200"/>
    <arg name="lidar_frame" default="laser"/>

    <!-- ========================= -->
    <!-- CobraFlex JSON Driver     -->
    <!-- ========================= -->
    <node
        pkg="cobraflex"
        exec="cobraflex_cmdvel_driver"
        name="cobraflex_cmdvel_driver"
        output="screen">

        <param name="port" value="$(var serial_port)"/>
        <param name="baud" value="$(var baudrate)"/>
        <param name="max_speed_value" value="100"/>
        <param name="max_linear" value="1.0"/>
        <param name="max_angular" value="1.0"/>
    </node>


    <!-- ========================= -->
    <!-- Lidar RPLIDAR A2M8        -->
    <!-- ========================= -->
    <include file="$(find-pkg-share rplidar_ros)/launch/view_rplidar_a2m8_launch.py">
        <arg name="frame_id" value="$(var lidar_frame)"/>
    </include>


    <!-- ========================= -->
    <!-- ZED Mini / ZED2 Camera    -->
    <!-- ========================= -->
    <include file="$(find-pkg-share zed_wrapper)/launch/zed2.launch.py">
        <!-- Puedes ajustar argumentos -->
        <!-- <arg name="resolution" value="HD720"/> -->
        <!-- <arg name="frame_rate" value="30"/> -->
    </include>


    <!-- ========================= -->
    <!-- Lidar Avoidance PID Node  -->
    <!-- ========================= -->
    <node
        pkg="cobraflex"
        exec="lidar_avoidance_pid"
        name="lidar_avoidance_pid"
        output="screen">
    </node>

</launch>
